lawnchair/src/app/lawnchair/search/LawnchairSearchAlgorithm.kt

```diff
@@ -92,6 +146,27 @@ sealed class LawnchairSearchAlgorithm(
             else -> LawnchairAppSearchAlgorithm(context)
         }
     }
+
+    protected suspend fun performDeviceWideSearch(query: String, prefs: PreferenceManager): MutableList<SearchResult> = withContext(Dispatchers.IO) {
+        val results = ArrayList<SearchResult>()
```

Comment:
Seems we can use `Sequence` here.

Reply:
yes we can when we implement max results option in separate commit